internal/driver/glfw/loop.go

```diff
@@ -97,94 +98,134 @@ func (d *gLDriver) runGL() {
 	settingsChange := make(chan fyne.Settings)
 	fyne.CurrentApp().Settings().AddChangeListener(settingsChange)
 
-	eventTick := time.NewTicker(time.Second / 60)
 	for {
-		select {
-		case <-d.done:
-			eventTick.Stop()
-			d.Terminate()
-			l := fyne.CurrentApp().Lifecycle().(*app.Lifecycle)
-			if f := l.OnStopped(); f != nil {
-				l.QueueEvent(f)
-			}
-			return
-		case f := <-funcQueue:
-			f.f()
-			f.done <- struct{}{}
-		case <-eventTick.C:
-			d.pollEvents()
-			windowsToRemove := 0
-			for _, win := range d.windowList() {
-				w := win.(*window)
-				if w.viewport == nil {
-					continue
+		d.waitEvents()
+
+		if d.animation.HasAnimations() {
+			// Switch to running at 60 fps with d.pollEvents
+			// until we have no more animations to tick
+			t := time.NewTicker(time.Second / 60)
```

Comment:
FYI @dweymouth: This is a memory leak for anyone using Go 1.22 or earlier, see https://pkg.go.dev/time#NewTicker.

Reply:
Updated

Reply:
Ah, never mind. You call stop on it everywhere. Is that fine?

Reply:
I suppose it probably is but it seems bound to fail us in the future when someone updates the code and misses a stop. Can defer work?

Reply:
Only if we create an anonymous func to encompass this little code block I guess. I suppose it's up to you, I prefer it as it is but I don't think it's important either way

Reply:
Someone not running Go 1.23 might get huge memory buildup if the app is running a long time. I think it is worth it but we can add a comment to remove the workaround code once 1.23 is our base.

Reply:
Updated